src/js/hooks/productSupplier/form/useProductSupplierForm.js

```diff
@@ -118,18 +144,30 @@ const useProductSupplierForm = () => {
       zodResolver(validationSchema(values))(values, context, options),
   });
 
+  useCalculateEachPrice({ control, setValue });
+
+  /* immediately trigger validation on preferenceType fields
+  when any values on defaultPreferenceType have changed */
+  const defaultPreferenceType = useWatch({ control, name: 'defaultPreferenceType' });
+  useEffect(() => {
+    trigger('defaultPreferenceType.preferenceType');
+  }, [defaultPreferenceType]);
+
   const onSubmit = (values) => {
     const payload = {
-      ...omitEmptyValues(values),
+      ...omitEmptyValues(values.basicDetails),
+      ...omitEmptyValues(values.additionalDetails),
+      ...omitEmptyValues(values.defaultPreferenceType),
+      ...omitEmptyValues(values.packageSpecification),
+      ...omitEmptyValues(values.fixedPrice),
```

Comment:
It could be better to use object restructuring. The `value` is repeated many times... but as I know the onSubmit will be refactored, so I don't know if this change is needed at this point.